L1Trigger/TrackFindingTMTT/src/TrkRZfilter.cc

```diff
@@ -0,0 +1,287 @@
+
+      // Declare this to be worth keeping (for now).
+      bool trackAccepted = true;
+
+      // Note that no estimate of the r-z helix params of this track yet exists.
+      rzHelix_set_ = false;  // No valid estimate yet.
+
+      // Digitize stubs for r-z filter if required.
+      if (settings_->enableDigitize()) {
+        for (Stub* s : stubs) {
+          s->digitize(iPhiSec_, Stub::DigiStage::SF);
+        }
+      }
+
+      //--- Filter stubs assigned to track, checking they are consistent with requested criteria.
+
+      // Get debug printout for specific regions.
+      bool print = false;
+
+      vector<Stub*> filteredStubs = stubs;
+      if (rzFilterName_ == "SeedFilter") {
+        filteredStubs = this->seedFilter(filteredStubs, trkIN.qOverPt(), print);
+      } else {
+        throw cms::Exception("BadConfig") << "TrkRzFilter: ERROR unknown r-z track filter requested: " << rzFilterName_;
+      }
+
+      // Check if track still has stubs in enough layers after filter.
+
+      unsigned int numLayersAfterFilters = Utility::countLayers(settings_, filteredStubs);
+      if (numLayersAfterFilters < settings_->minFilterLayers())
+        trackAccepted = false;
+      //if (numLayersAfterFilters < Utility::numLayerCut(Utility::AlgoStep::SEED, settings_, iPhiSec_, iEtaReg_, std::abs(trkIN.qOverPt()))) trackAccepted = false;
+
+      if (trackAccepted) {
+        // Estimate r-z helix parameters from centre of eta-sector if no estimate provided by r-z filter.
+        if (!rzHelix_set_)
+          this->estRZhelix();
+
+        pair<float, float> helixRZ(rzHelix_z0_, rzHelix_tanL_);
+
+        // Create copy of original track, except now using its filtered stubs, to be added to filteredTrack collection.
+        filteredTracks.emplace_back(settings_,
+                                    filteredStubs,
+                                    trkIN.cellLocationHT(),
+                                    trkIN.helix2D(),
+                                    helixRZ,
+                                    trkIN.iPhiSec(),
+                                    trkIN.iEtaReg(),
+                                    trkIN.optoLinkID(),
+                                    trkIN.mergedHTcell());
+      }
+    }
+
+    return filteredTracks;
+  }
+
+  //=== Use Seed Filter to produce a filtered collection of stubs on this track candidate that are consistent with a straight line
+  //=== in r-z using tracklet algo.
+
+  vector<Stub*> TrkRZfilter::seedFilter(const std::vector<Stub*>& stubs, float trkQoverPt, bool print) {
+    unsigned int numLayers;                //Num of Layers in the cell after that filter has been applied
+    std::vector<Stub*> filtStubs = stubs;  // Copy stubs vector in filtStubs
+    bool FirstSeed = true;
+    //Allowed layers for the first & second seeding stubs
+    static const std::vector<unsigned int> FirstSeedLayers = {1, 2, 11, 21, 3, 12, 22, 4};
```

Comment:
remaining static variables (probably harmless)

Reply:
Since "static const" data is thread safe, and following the discussion leading to https://github.com/cms-sw/cmssw/pull/29381#issuecomment-628889782 , we propose to leave this code unchanged.